src/docfx/template/TemplateEngine.cs

```diff
@@ -18,16 +17,13 @@ internal class TemplateEngine
     {
         private const string DefaultTemplateDir = "_themes";
         private static readonly string[] s_resourceFolders = new[] { "global", "css", "fonts" };
-        private static readonly ConcurrentDictionary<string, Lazy<(JsonSchemaValidator, JsonSchemaTransformer)>> _jsonSchemas
-                          = new ConcurrentDictionary<string, Lazy<(JsonSchemaValidator, JsonSchemaTransformer)>>();
 
         private readonly string _templateDir;
-        private readonly string _schemaDir;
         private readonly LiquidTemplate _liquid;
         private readonly JavascriptEngine _js;
         private readonly HashSet<string> _htmlMetaHidden;
         private readonly Dictionary<string, string> _htmlMetaNames;
-        private readonly HashSet<string> _schemas;
+        private readonly IReadOnlyDictionary<string, TemplateSchema> _schemas;
```

Comment:
Move Lazies inside `TemplateSchema` over here:
```suggestion
        private readonly IReadOnlyDictionary<string, Lazy<TemplateSchema>> _schemas;
```